internal/armada/server/handlers/submit_jobs_test.go

```diff
@@ -0,0 +1,282 @@
+			)
+
+			_, err := handler(context.Background(), &request)
+			if err != nil {
+				t.Errorf("failed to handle request: %s", err)
+			}
+
+			return true
+		},
+	}
+
+	for name, property := range properties {
+		t.Run(name, func(tb *testing.T) {
+			if err := quick.Check(property, nil); err != nil {
+				tb.Error(err)
+			}
+		})
+	}
+
+}
+
+func TestSubmitJobsValidate(t *testing.T) {
+	properties := map[string]interface{}{
+		"notFound": func(request api.JobSubmitRequest) bool {
+			if len(request.JobRequestItems) == 0 {
+				return true
+			}
+
+			handler := submitJobs(func(c context.Context, jsr *api.JobSubmitRequest) (*api.JobSubmitResponse, error) {
+				return nil, nil
+			}).Validate(func(item *api.JobSubmitRequestItem) error {
+				return fmt.Errorf("")
+			})
+
+			_, err := handler(context.Background(), &request)
+			if err == nil {
+				t.Errorf("failed to handle an error")
+				return false
+			}
+
+			if status, _ := status.FromError(err); status.Code() != codes.InvalidArgument {
+				t.Errorf("invalid status code: %s", status.Code())
+				return false
+			}
+
+			return true
+		},
+		"passthrough": func(request api.JobSubmitRequest, response api.JobSubmitResponse) bool {
+			request.Queue += "a"
+			request.JobSetId += "a"
+
+			handler := submitJobs(func(c context.Context, jsr *api.JobSubmitRequest) (*api.JobSubmitResponse, error) {
+				if !reflect.DeepEqual(&request, jsr) {
+					return nil, fmt.Errorf("invalid request data")
+				}
+				return &response, nil
+			}).Validate(func(item *api.JobSubmitRequestItem) error {
+				return nil
+			})
+
+			out, err := handler(context.Background(), &request)
+			if err != nil {
+				t.Errorf("failed to handle request: %s", err)
+			}
+
+			return reflect.DeepEqual(&response, out)
+		},
+	}
+
+	for name, property := range properties {
+		t.Run(name, func(tb *testing.T) {
+			if err := quick.Check(property, nil); err != nil {
+				tb.Error(err)
+			}
+		})
+	}
+}
+
+func TestSubmitJobsAuthorize(t *testing.T) {
+	properties := map[string]interface{}{
+		"notFound": func(request api.JobSubmitRequest, sumbitAnyJob bool) bool {
+			handler := submitJobs(func(c context.Context, jsr *api.JobSubmitRequest) (*api.JobSubmitResponse, error) {
+				return nil, nil
+			}).Authorize(
+				func(ctx context.Context, queueName string) (bool, error) {
+					return false, repository.ErrQueueNotFound
+				},
+				func(c context.Context, p permission.Permission) bool {
+					if sumbitAnyJob {
+						return p == permissions.SubmitAnyJobs
+					}
+					return p == permissions.SubmitJobs
+				},
+				false,
+			)
+
+			_, err := handler(context.Background(), &request)
+			if err == nil {
+				t.Errorf("failed to handle an error")
+				return false
+			}
+
+			if status, _ := status.FromError(err); status.Code() != codes.NotFound {
+				t.Errorf("invalid status code: %s", status.Code())
+				return false
+			}
+
+			return true
+		},
+		"internal": func(request api.JobSubmitRequest, sumbitAnyJob bool) bool {
```

Comment:
clearer name for this test would be good

Reply:
Hm, It's the name of the status code returned by handler. Do you have any propositions?